class_name WireEntity
extends Entity

@onready var sprites: Array[Sprite2D] = [$SpriteRight, $SpriteDown, $SpriteLeft, $SpriteUp]
@onready var _flow_animations: Array[AnimatedSprite2D] = [
	$AnimationRight, $AnimationDown, $AnimationLeft, $AnimationUp
]


func _ready():
	pass  # Replace with function body.


func _on_power_input(amount, _delta, direction: Vector2):
	if amount <= 0:
		return

	var animation: AnimatedSprite2D = _get_animation_from(direction)
	if animation != null:
		animation.play("input")


func _on_power_output(amount, _delta, direction: Vector2):
	if amount <= 0:
		return

	var animation: AnimatedSprite2D = _get_animation_from(direction)
	if animation != null:
		animation.play("output")


func _get_animation_from(direction: Vector2):
	match direction:
		Vector2.RIGHT:
			return _flow_animations[0]
		Vector2.DOWN:
			return _flow_animations[1]
		Vector2.LEFT:
			return _flow_animations[2]
		Vector2.UP:
			return _flow_animations[3]
	return null
